shop/src/main/java/ro/msg/learning/shop/Configurations/StrategyConfiguration.java

```diff
@@ -0,0 +1,27 @@
+package ro.msg.learning.shop.Configurations;
+
+import lombok.AllArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import ro.msg.learning.shop.Strategies.MostAbundantStrategyConfiguration;
+import ro.msg.learning.shop.Strategies.SingleLocationStrategyConfiguration;
+import ro.msg.learning.shop.Strategies.StrategyConfigurationInterface;
+
+@Configuration
+@AllArgsConstructor
+public class StrategyConfiguration {
+
+    @Bean
+    public StrategyConfigurationInterface chooseStrategy(@Value("${strategy}")String s) throws Exception {
+        if(s.equals("SingleLocationStrategy")){
```

Comment:
You can create an Enum for Strategy Type and replace the hardcoded text with Enum value

Reply:
Fixed.